While doing rollingupdate weave-npc panics and restarts

While doing a rollingupdate I am seeing weave-npc throwing a panic on the legacy clusters.  I am running weave 1.8.2.

1. create a cluster
2. drain, cordon, destroy node
3. weave-net on existing nodes panics

```
E0208 00:38:22.635219    3164 runtime.go:64] Observed a panic: &runtime.TypeAssertionError{interfaceString:"interface {}", concreteString:"cache.DeletedFinalStateUnknown", assertedString:"*v1.Pod", missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod)
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:70
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:63
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:49
/usr/local/go/src/runtime/asm_amd64.s:437
/usr/local/go/src/runtime/panic.go:423
/usr/local/go/src/runtime/iface.go:175
/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:142
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:182
<autogenerated>:52
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:252
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:126
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102
/usr/local/go/src/runtime/asm_amd64.s:1721
E0208 00:38:22.635392    3164 runtime.go:64] Observed a panic: &runtime.TypeAssertionError{interfaceString:"interface {}", concreteString:"cache.DeletedFinalStateUnknown", assertedString:"*v1.Pod", missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod)
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:70
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:63
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:49
/usr/local/go/src/runtime/asm_amd64.s:437
/usr/local/go/src/runtime/panic.go:423
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:56
/usr/local/go/src/runtime/asm_amd64.s:437
/usr/local/go/src/runtime/panic.go:423
/usr/local/go/src/runtime/iface.go:175
/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:142
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:182
<autogenerated>:52
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:252
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:126
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49
/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102
/usr/local/go/src/runtime/asm_amd64.s:1721
panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod [recovered]
	panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod [recovered]
	panic: interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.Pod

goroutine 30 [running]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:56 +0x153
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:56 +0x153
main.root.func5(0x19f4220, 0xc8217291c0)
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:142 +0x92
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnDelete(0xc82160b500, 0xc82160b540, 0xc82160b520, 0x19f4220, 0xc8217291c0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:182 +0x3a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*ResourceEventHandlerFuncs).OnDelete(0xc82160b580, 0x19f4220, 0xc8217291c0)
	<autogenerated>:52 +0xb5
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.NewInformer.func1(0x19f43e0, 0xc821f57300, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:252 +0x56d
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc82007a6e0, 0xc82162f800, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420 +0x29f
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).processLoop(0xc820338cb0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:126 +0x4a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).(github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.processLoop)-fm()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x20
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc82165ff58)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc82165ff58, 0x3b9aca00, 0x0, 0x163cc01, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc82165ff58, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).Run(0xc820338cb0, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x1c2
created by main.root
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:161 +0xf13

goroutine 1 [chan receive, 14 minutes]:
main.root(0xc8201cafc0, 0x2ffbcc0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:166 +0x111e
github.com/weaveworks/weave/vendor/github.com/spf13/cobra.(*Command).execute(0xc8201cafc0, 0xc82000a1f0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/github.com/spf13/cobra/command.go:636 +0x8f8
github.com/weaveworks/weave/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc8201cafc0, 0xc8201cafc0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/github.com/spf13/cobra/command.go:722 +0x56b
github.com/weaveworks/weave/vendor/github.com/spf13/cobra.(*Command).Execute(0xc8201cafc0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/github.com/spf13/cobra/command.go:681 +0x2d
main.main()
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:177 +0x107

goroutine 17 [syscall, 14 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 5 [syscall, 14 minutes]:
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [chan receive, 2 minutes]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x2a9fe60)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/vendor/github.com/golang/glog/glog.go:879 +0x67
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/vendor/github.com/golang/glog.init.1
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/vendor/github.com/golang/glog/glog.go:410 +0x297

goroutine 14 [IO wait, 14 minutes]:
net.runtime_pollWait(0x7fd2ac436228, 0x72, 0xc82000a110)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8204175d0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8204175d0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820417570, 0x0, 0x7fd2ac436320, 0xc821543da0)
	/usr/local/go/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820466170, 0xc82158bd50, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:254 +0x4d
net/http.tcpKeepAliveListener.Accept(0xc820466170, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:2135 +0x41
net/http.(*Server).Serve(0xc8204d4660, 0x7fd2ac4362e8, 0xc820466170, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1887 +0xb3
net/http.(*Server).ListenAndServe(0xc8204d4660, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1877 +0x136
net/http.ListenAndServe(0x1c46d30, 0x5, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1967 +0x8f
github.com/weaveworks/weave/npc/metrics.Start.func1(0x1c46d30, 0x5)
	/go/src/github.com/weaveworks/weave/npc/metrics/metrics.go:102 +0x131
created by github.com/weaveworks/weave/npc/metrics.Start
	/go/src/github.com/weaveworks/weave/npc/metrics/metrics.go:105 +0xe4

goroutine 15 [syscall, 14 minutes]:
syscall.Syscall(0x0, 0x7, 0xc82158e000, 0x10, 0x7fd200000000, 0x100000000000001, 0x102aa4a40)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x7, 0xc82158e000, 0x10, 0x10010, 0x7fd2ac436348, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x7, 0xc82158e000, 0x10, 0x10010, 0x1b13fc0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc820466178, 0xc82158e000, 0x10, 0x10010, 0x163d180, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc820466178, 0xc82158e000, 0x10, 0x10010, 0xc8203bd570, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
io.ReadAtLeast(0x7fd2ae58d270, 0xc820466178, 0xc82158e000, 0x10, 0x10010, 0x10, 0x0, 0x0, 0x0)
	/usr/local/go/src/io/io.go:298 +0xe6
io.ReadFull(0x7fd2ae58d270, 0xc820466178, 0xc82158e000, 0x10, 0x10010, 0x452810, 0x0, 0x0)
	/usr/local/go/src/io/io.go:316 +0x62
github.com/weaveworks/weave/vendor/github.com/google/gopacket/pcapgo.(*Reader).readPacketHeader(0xc8203bd540, 0x0, 0xc800000000, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/github.com/google/gopacket/pcapgo/read.go:119 +0xbe
github.com/weaveworks/weave/vendor/github.com/google/gopacket/pcapgo.(*Reader).ReadPacketData(0xc8203bd540, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/weave/vendor/github.com/google/gopacket/pcapgo/read.go:99 +0x8b
github.com/weaveworks/weave/npc/metrics.gatherMetrics()
	/go/src/github.com/weaveworks/weave/npc/metrics/metrics.go:38 +0x296
created by github.com/weaveworks/weave/npc/metrics.Start
	/go/src/github.com/weaveworks/weave/npc/metrics/metrics.go:107 +0xfc

goroutine 16 [syscall, 14 minutes]:
syscall.Syscall(0x0, 0x3, 0xc821586000, 0x8000, 0x4676b8, 0x1a3a6e0, 0x0)
	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
syscall.read(0x3, 0xc821586000, 0x8000, 0x8000, 0x4106a5, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:783 +0x5f
syscall.Read(0x3, 0xc821586000, 0x8000, 0x8000, 0xc82001be08, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_unix.go:160 +0x4d
os.(*File).read(0xc820466110, 0xc821586000, 0x8000, 0x8000, 0x6d, 0x0, 0x0)
	/usr/local/go/src/os/file_unix.go:211 +0x53
os.(*File).Read(0xc820466110, 0xc821586000, 0x8000, 0x8000, 0x6d, 0x0, 0x0)
	/usr/local/go/src/os/file.go:95 +0x8a
io.copyBuffer(0x7fd2ae58d1c0, 0xc820022010, 0x7fd2ae58d270, 0xc820466110, 0xc821586000, 0x8000, 0x8000, 0x6d, 0x0, 0x0)
	/usr/local/go/src/io/io.go:381 +0x247
io.Copy(0x7fd2ae58d1c0, 0xc820022010, 0x7fd2ae58d270, 0xc820466110, 0x977f1c, 0x0, 0x0)
	/usr/local/go/src/io/io.go:351 +0x64
created by github.com/weaveworks/weave/npc/ulogd.Start
	/go/src/github.com/weaveworks/weave/npc/ulogd/ulogd.go:26 +0x1d2

goroutine 18 [syscall, 14 minutes]:
syscall.Syscall6(0x3d, 0xc92, 0xc82001c56c, 0x0, 0xc82006ee10, 0x0, 0x0, 0x4526b0, 0x7fd2aef50000, 0xa)
	/usr/local/go/src/syscall/asm_linux_amd64.s:44 +0x5
syscall.wait4(0xc92, 0xc82001c56c, 0x0, 0xc82006ee10, 0x90, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:172 +0x72
syscall.Wait4(0xc92, 0xc82001c5b4, 0x0, 0xc82006ee10, 0x0, 0x0, 0x0)
	/usr/local/go/src/syscall/syscall_linux.go:256 +0x55
os.(*Process).wait(0xc821542da0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/exec_unix.go:22 +0x105
os.(*Process).Wait(0xc821542da0, 0x0, 0x0, 0x0)
	/usr/local/go/src/os/doc.go:45 +0x2d
os/exec.(*Cmd).Wait(0xc82008b540, 0x0, 0x0)
	/usr/local/go/src/os/exec/exec.go:380 +0x211
github.com/weaveworks/weave/npc/ulogd.waitForExit(0xc82008b540)
	/go/src/github.com/weaveworks/weave/npc/ulogd/ulogd.go:11 +0x3c
created by github.com/weaveworks/weave/npc/ulogd.Start
	/go/src/github.com/weaveworks/weave/npc/ulogd/ulogd.go:27 +0x1f4

goroutine 29 [semacquire, 14 minutes]:
sync.runtime_Syncsemacquire(0xc82007a658)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82007a648)
	/usr/local/go/src/sync/cond.go:62 +0x9b
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc82007a630, 0xc82162f6e0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:407 +0xc4
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).processLoop(0xc820338c40)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:126 +0x4a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).(github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.processLoop)-fm()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x20
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc82002ff88)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc82002ff88, 0x3b9aca00, 0x0, 0x163cc01, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc82002ff88, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).Run(0xc820338c40, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x1c2
created by main.root
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:160 +0xee5

goroutine 31 [semacquire, 14 minutes]:
sync.runtime_Syncsemacquire(0xc82007a7b8)
	/usr/local/go/src/runtime/sema.go:237 +0x201
sync.(*Cond).Wait(0xc82007a7a8)
	/usr/local/go/src/sync/cond.go:62 +0x9b
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*DeltaFIFO).Pop(0xc82007a790, 0xc82162f920, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:407 +0xc4
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).processLoop(0xc820338d20)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:126 +0x4a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).(github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.processLoop)-fm()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x20
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc820033f88)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc820033f88, 0x3b9aca00, 0x0, 0x163cc01, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc820033f88, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Controller).Run(0xc820338d20, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/controller.go:102 +0x1c2
created by main.root
	/go/src/github.com/weaveworks/weave/prog/weave-npc/main.go:162 +0xf41

goroutine 32 [select, 14 minutes, locked to thread]:
runtime.gopark(0x201ebf0, 0xc82001af28, 0x1c543e8, 0x6, 0xc8202aec18, 0x2)
	/usr/local/go/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc82001af28, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc82001af28)
	/usr/local/go/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1721 +0x1

goroutine 33 [runnable]:
net/http.(*Transport).getConn(0xc82006ed80, 0xc821dbc9a0, 0x0, 0xc82214e7c0, 0x5, 0xc82214e7c8, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:554 +0x663
net/http.(*Transport).RoundTrip(0xc82006ed80, 0xc821dbc9a0, 0xd, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:228 +0x4ac
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*bearerAuthRoundTripper).RoundTrip(0xc82153d420, 0xc821dbc9a0, 0xa, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:183 +0x23e
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*userAgentRoundTripper).RoundTrip(0xc82153d6e0, 0xc821dbc7e0, 0x2, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:92 +0x130
net/http.send(0xc821dbc700, 0x7fd2ac436580, 0xc82153d6e0, 0xc82214e800, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:220 +0x52c
net/http.(*Client).send(0xc821535980, 0xc821dbc700, 0x3a, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:143 +0x15a
net/http.(*Client).doFollowingRedirects(0xc821535980, 0xc821dbc700, 0x201e7b8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:380 +0xbc0
net/http.(*Client).Do(0xc821535980, 0xc821dbc700, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:175 +0x9c
github.com/weaveworks/weave/vendor/k8s.io/client-go/rest.(*Request).request(0xc820276680, 0xc8219758c8, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/rest/request.go:804 +0x601
github.com/weaveworks/weave/vendor/k8s.io/client-go/rest.(*Request).Do(0xc820276680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/rest/request.go:863 +0x85
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.NewListWatchFromClient.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1c46a60, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:56 +0x1ff
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*ListWatch).List(0xc8214c6c50, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:80 +0x5f
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc82032e980, 0xc8200151a0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:258 +0x304
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil.func1()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:213 +0x29
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc82160b680)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc82160b680, 0x3b9aca00, 0x0, 0xc8201eec01, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc82160b680, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:216 +0x276

goroutine 48 [select, 14 minutes]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc8204663b8, 0xc8200151a0, 0xc82032f380, 0xc8200fb200, 0xc8204663c0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:279 +0x321
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:292 +0xddd

goroutine 35 [select]:
net/http.(*Transport).getConn(0xc82006ed80, 0xc821736d20, 0x0, 0xc821d7a910, 0x5, 0xc821d7a918, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:554 +0x663
net/http.(*Transport).RoundTrip(0xc82006ed80, 0xc821736d20, 0xd, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:228 +0x4ac
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*bearerAuthRoundTripper).RoundTrip(0xc82153d420, 0xc821736d20, 0xa, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:183 +0x23e
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*userAgentRoundTripper).RoundTrip(0xc82153d6e0, 0xc821736b60, 0xa7, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:92 +0x130
net/http.send(0xc821736a80, 0x7fd2ac436580, 0xc82153d6e0, 0xc821d7a960, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:220 +0x52c
net/http.(*Client).send(0xc821535980, 0xc821736a80, 0x50, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:143 +0x15a
net/http.(*Client).doFollowingRedirects(0xc821535980, 0xc821736a80, 0x201e7b8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:380 +0xbc0
net/http.(*Client).Do(0xc821535980, 0xc821736a80, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:175 +0x9c
github.com/weaveworks/weave/vendor/k8s.io/client-go/rest.(*Request).Watch(0xc8202771e0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/rest/request.go:659 +0x42a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.NewListWatchFromClient.func2(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc8214ecd70, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:66 +0x22c
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*ListWatch).Watch(0xc8214c6c70, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:85 +0x5f
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc82032ed00, 0xc8200151a0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:303 +0xf31
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil.func1()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:213 +0x29
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc82160ba00)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc82160ba00, 0x3b9aca00, 0x0, 0xc8200dc501, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc82160ba00, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:216 +0x276

goroutine 113 [select]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc820466040, 0xc8200151a0, 0xc82032ed00, 0xc821e47ce0, 0xc820466048)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:279 +0x321
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:292 +0xddd

goroutine 37 [select]:
net/http.(*Transport).getConn(0xc82006ed80, 0xc8216e7420, 0x0, 0xc8202d3dc0, 0x5, 0xc8202d3dc8, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:554 +0x663
net/http.(*Transport).RoundTrip(0xc82006ed80, 0xc8216e7420, 0xd, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:228 +0x4ac
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*bearerAuthRoundTripper).RoundTrip(0xc8214f13e0, 0xc8216e7420, 0xa, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:183 +0x23e
github.com/weaveworks/weave/vendor/k8s.io/client-go/transport.(*userAgentRoundTripper).RoundTrip(0xc8214f1440, 0xc8216e7340, 0xae, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/transport/round_trippers.go:92 +0x130
net/http.send(0xc8216e7260, 0x7fd2ac436580, 0xc8214f1440, 0xc820252460, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:220 +0x52c
net/http.(*Client).send(0xc821540330, 0xc8216e7260, 0x6c, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:143 +0x15a
net/http.(*Client).doFollowingRedirects(0xc821540330, 0xc8216e7260, 0x201e7b8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:380 +0xbc0
net/http.(*Client).Do(0xc821540330, 0xc8216e7260, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:175 +0x9c
github.com/weaveworks/weave/vendor/k8s.io/client-go/rest.(*Request).Watch(0xc820277860, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/rest/request.go:659 +0x42a
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.NewListWatchFromClient.func2(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc821364180, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:66 +0x22c
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*ListWatch).Watch(0xc8214c6c80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/listwatch.go:85 +0x5f
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch(0xc82032f380, 0xc8200151a0, 0x0, 0x0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:303 +0xf31
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil.func1()
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:213 +0x29
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil.func1(0xc82160bc80)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:87 +0x4b
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.JitterUntil(0xc82160bc80, 0x3b9aca00, 0x0, 0xc8201eed01, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:88 +0xb4
github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait.Until(0xc82160bc80, 0x3b9aca00, 0xc8200151a0)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/pkg/util/wait/wait.go:49 +0x43
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).RunUntil
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:216 +0x276

goroutine 110 [runnable]:
net.runtime_pollWait(0x7fd2ac435fe8, 0x72, 0xc82000a110)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820444ca0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820444ca0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820444c40, 0xc821792000, 0x400, 0x400, 0x0, 0x7fd2ae589050, 0xc82000a110)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820466128, 0xc821792000, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc822147f80, 0x7fd2ac436f78, 0xc820466128, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82204a580, 0x14, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*clientHandshakeState).readFinished(0xc821e95e48, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/handshake_client.go:559 +0x55
crypto/tls.(*Conn).clientHandshake(0xc82204a580, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/handshake_client.go:218 +0x17da
crypto/tls.(*Conn).Handshake(0xc82204a580, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:974 +0xec
net/http.(*Transport).dialConn.func3(0xc82204a580, 0xc82214e880, 0xc821e47140)
	/usr/local/go/src/net/http/transport.go:662 +0x21
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:667 +0x7bc

goroutine 115 [chan receive]:
net/http.(*Transport).getConn.func2.1(0xc821e46ae0, 0xc82006ed80, 0x0)
	/usr/local/go/src/net/http/transport.go:536 +0x4c
created by net/http.(*Transport).getConn.func2
	/usr/local/go/src/net/http/transport.go:542 +0x77

goroutine 112 [select]:
net/http.(*persistConn).writeLoop(0xc821da00b0)
	/usr/local/go/src/net/http/transport.go:1009 +0x40c
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:686 +0xc9d

goroutine 109 [runnable]:
net.runtime_pollWait(0x7fd2ac4360a8, 0x72, 0xc82000a110)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820444a70, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820444a70, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820444a10, 0xc821763000, 0x400, 0x400, 0x0, 0x7fd2ae589050, 0xc82000a110)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820466108, 0xc821763000, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8218906c0, 0x7fd2ac436f78, 0xc820466108, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82204a2c0, 0x14, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*clientHandshakeState).readFinished(0xc821b4fe48, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/handshake_client.go:559 +0x55
crypto/tls.(*Conn).clientHandshake(0xc82204a2c0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/handshake_client.go:218 +0x17da
crypto/tls.(*Conn).Handshake(0xc82204a2c0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:974 +0xec
net/http.(*Transport).dialConn.func3(0xc82204a2c0, 0xc82214e840, 0xc821e470e0)
	/usr/local/go/src/net/http/transport.go:662 +0x21
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:667 +0x7bc

goroutine 114 [chan receive]:
net/http.(*Transport).dialConn(0xc82006ed80, 0x0, 0xc821d7a910, 0x5, 0xc821d7a918, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:668 +0x7f8
net/http.(*Transport).getConn.func4(0xc82006ed80, 0x0, 0xc821d7a910, 0x5, 0xc821d7a918, 0xe, 0xc821e47e60)
	/usr/local/go/src/net/http/transport.go:549 +0x66
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265

goroutine 49 [select, 14 minutes]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc8204663a0, 0xc8200151a0, 0xc82032ed00, 0xc820014420, 0xc8204663a8)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:279 +0x321
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:292 +0xddd

goroutine 51 [select, 14 minutes]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc820466380, 0xc8200151a0, 0xc82032e980, 0xc82015d020, 0xc820466388)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:279 +0x321
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:292 +0xddd

goroutine 111 [IO wait]:
net.runtime_pollWait(0x7fd2ac435f28, 0x72, 0xc82000a110)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820444220, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820444220, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8204441c0, 0xc821a9e000, 0x8000, 0x8000, 0x0, 0x7fd2ae589050, 0xc82000a110)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820466060, 0xc821a9e000, 0x8000, 0x8000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8221468a0, 0x7fd2ac436f78, 0xc820466060, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc82204a000, 0x201ec17, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).Read(0xc82204a000, 0xc8200d6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:901 +0x167
net/http.noteEOFReader.Read(0x7fd2ac437388, 0xc82204a000, 0xc821da0108, 0xc8200d6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1370 +0x67
net/http.(*noteEOFReader).Read(0xc822195840, 0xc8200d6000, 0x1000, 0x1000, 0x2, 0x0, 0x0)
	<autogenerated>:126 +0xd0
bufio.(*Reader).fill(0xc821e47260)
	/usr/local/go/src/bufio/bufio.go:97 +0x1e9
bufio.(*Reader).Peek(0xc821e47260, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0xcc
net/http.(*persistConn).readLoop(0xc821da00b0)
	/usr/local/go/src/net/http/transport.go:876 +0xf7
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:685 +0xc78

goroutine 108 [chan receive]:
net/http.(*Transport).dialConn(0xc82006ed80, 0x0, 0xc82214e7c0, 0x5, 0xc82214e7c8, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:668 +0x7f8
net/http.(*Transport).getConn.func4(0xc82006ed80, 0x0, 0xc82214e7c0, 0x5, 0xc82214e7c8, 0xe, 0xc821e46e40)
	/usr/local/go/src/net/http/transport.go:549 +0x66
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265

goroutine 107 [chan receive]:
net/http.(*Transport).dialConn(0xc82006ed80, 0x0, 0xc822099220, 0x5, 0xc822099228, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:668 +0x7f8
net/http.(*Transport).getConn.func4(0xc82006ed80, 0x0, 0xc822099220, 0x5, 0xc822099228, 0xe, 0xc821e46ae0)
	/usr/local/go/src/net/http/transport.go:549 +0x66
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265

goroutine 116 [runnable]:
net.runtime_pollWait(0x7fd2ac436168, 0x72, 0xc82000a110)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8203316b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8203316b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820331650, 0xc821763c00, 0x400, 0x400, 0x0, 0x7fd2ae589050, 0xc82000a110)
	/usr/local/go/src/net/fd_unix.go:232 +0x23a
net.(*conn).Read(0xc820466148, 0xc821763c00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
crypto/tls.(*block).readFromUntil(0xc8218909f0, 0x7fd2ac436f78, 0xc820466148, 0x5, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:455 +0xcc
crypto/tls.(*Conn).readRecord(0xc821a64dc0, 0x16, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:540 +0x2d1
crypto/tls.(*Conn).readHandshake(0xc821a64dc0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:776 +0xd3
crypto/tls.(*Conn).clientHandshake(0xc821a64dc0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/handshake_client.go:142 +0xf77
crypto/tls.(*Conn).Handshake(0xc821a64dc0, 0x0, 0x0)
	/usr/local/go/src/crypto/tls/conn.go:974 +0xec
net/http.(*Transport).dialConn.func3(0xc821a64dc0, 0xc821d65500, 0xc821c05200)
	/usr/local/go/src/net/http/transport.go:662 +0x21
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:667 +0x7bc

goroutine 117 [runnable]:
github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch.func1(0xc820466070, 0xc8200151a0, 0xc82032f380, 0xc8200fb920, 0xc820466080)
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:277
created by github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache.(*Reflector).ListAndWatch
	/go/src/github.com/weaveworks/weave/vendor/k8s.io/client-go/tools/cache/reflector.go:292 +0xddd

goroutine 118 [runnable]:
net.runtime_pollWait(0x7fd2ac435e68, 0x77, 0x453ff0)
	/usr/local/go/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8202521b0, 0x77, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitWrite(0xc8202521b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:82 +0x36
net.(*netFD).connect(0xc820252150, 0x0, 0x0, 0x7fd2ac436f50, 0xc821e96f40, 0xed02c5f9c, 0x24806610, 0x2a9fb00, 0x0, 0x0)
	/usr/local/go/src/net/fd_unix.go:114 +0x1f6
net.(*netFD).dial(0xc820252150, 0x7fd2ac4352e0, 0x0, 0x7fd2ac4352e0, 0xc8216eb920, 0xed02c5f9c, 0x24806610, 0x2a9fb00, 0x0, 0x0)
	/usr/local/go/src/net/sock_posix.go:137 +0x351
net.socket(0x1c553c0, 0x3, 0x2, 0x1, 0x0, 0xc8216eb900, 0x7fd2ac4352e0, 0x0, 0x7fd2ac4352e0, 0xc8216eb920, ...)
	/usr/local/go/src/net/sock_posix.go:89 +0x411
net.internetSocket(0x1c553c0, 0x3, 0x7fd2ac4352e0, 0x0, 0x7fd2ac4352e0, 0xc8216eb920, 0xed02c5f9c, 0xc824806610, 0x2a9fb00, 0x1, ...)
	/usr/local/go/src/net/ipsock_posix.go:160 +0x141
net.dialTCP(0x1c553c0, 0x3, 0x0, 0xc8216eb920, 0xed02c5f9c, 0xc824806610, 0x2a9fb00, 0xc8213645f0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:171 +0x11e
net.dialSingle(0xc8202f1000, 0x7fd2ac435278, 0xc8216eb920, 0xed02c5f9c, 0x24806610, 0x2a9fb00, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:364 +0x3f5
net.dialSerial.func1(0xed02c5f9c, 0x24806610, 0x2a9fb00, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:336 +0x75
net.dial(0x1c553c0, 0x3, 0x7fd2ac435278, 0xc8216eb920, 0xc8219796e8, 0xed02c5f9c, 0x24806610, 0x2a9fb00, 0x0, 0x0, ...)
	/usr/local/go/src/net/fd_unix.go:40 +0x60
net.dialSerial(0xc8202f1000, 0xc8213645f0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:338 +0x760
net.(*Dialer).Dial(0xc8203bd360, 0x1c553c0, 0x3, 0xc8202d3dc8, 0xe, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/dial.go:232 +0x50f
net.(*Dialer).Dial-fm(0x1c553c0, 0x3, 0xc8202d3dc8, 0xe, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:38 +0x6e
net/http.(*Transport).dial(0xc82006ed80, 0x1c553c0, 0x3, 0xc8202d3dc8, 0xe, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:499 +0x79
net/http.(*Transport).dialConn(0xc82006ed80, 0x0, 0xc8202d3dc0, 0x5, 0xc8202d3dc8, 0xe, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:596 +0x19a9
net/http.(*Transport).getConn.func4(0xc82006ed80, 0x0, 0xc8202d3dc0, 0x5, 0xc8202d3dc8, 0xe, 0xc82015d320)
	/usr/local/go/src/net/http/transport.go:549 +0x66
created by net/http.(*Transport).getConn
	/usr/local/go/src/net/http/transport.go:551 +0x265
```

--- a/prog/weave-npc/main.go
+++ b/prog/weave-npc/main.go
@@ -127,7 +127,15 @@ func root(cmd *cobra.Command, args []string) {
 				handleError(npc.AddNamespace(obj.(*coreapi.Namespace)))
 			},
 			DeleteFunc: func(obj interface{}) {
-				handleError(npc.DeleteNamespace(obj.(*coreapi.Namespace)))
+				switch obj := obj.(type) {
+				case *coreapi.Namespace:
+					handleError(npc.DeleteNamespace(obj))
+				case cache.DeletedFinalStateUnknown:
+					// We know this object has gone away, but its final state is no longer
+					// available from the API server. Instead we use the last copy of it
+					// that we have, which is good enough for our cleanup.
+					handleError(npc.DeleteNamespace(obj.Obj.(*coreapi.Namespace)))
+				}
 			},
 			UpdateFunc: func(old, new interface{}) {
 				handleError(npc.UpdateNamespace(old.(*coreapi.Namespace), new.(*coreapi.Namespace)))
@@ -139,7 +147,15 @@ func root(cmd *cobra.Command, args []string) {
 				handleError(npc.AddPod(obj.(*coreapi.Pod)))
 			},
 			DeleteFunc: func(obj interface{}) {
-				handleError(npc.DeletePod(obj.(*coreapi.Pod)))
+				switch obj := obj.(type) {
+				case *coreapi.Pod:
+					handleError(npc.DeletePod(obj))
+				case cache.DeletedFinalStateUnknown:
+					// We know this object has gone away, but its final state is no longer
+					// available from the API server. Instead we use the last copy of it
+					// that we have, which is good enough for our cleanup.
+					handleError(npc.DeletePod(obj.Obj.(*coreapi.Pod)))
+				}
 			},
 			UpdateFunc: func(old, new interface{}) {
 				handleError(npc.UpdatePod(old.(*coreapi.Pod), new.(*coreapi.Pod)))
@@ -151,7 +167,15 @@ func root(cmd *cobra.Command, args []string) {
 				handleError(npc.AddNetworkPolicy(obj.(*extnapi.NetworkPolicy)))
 			},
 			DeleteFunc: func(obj interface{}) {
-				handleError(npc.DeleteNetworkPolicy(obj.(*extnapi.NetworkPolicy)))
+				switch obj := obj.(type) {
+				case *extnapi.NetworkPolicy:
+					handleError(npc.DeleteNetworkPolicy(obj))
+				case cache.DeletedFinalStateUnknown:
+					// We know this object has gone away, but its final state is no longer
+					// available from the API server. Instead we use the last copy of it
+					// that we have, which is good enough for our cleanup.
+					handleError(npc.DeleteNetworkPolicy(obj.Obj.(*extnapi.NetworkPolicy)))
+				}
 			},
 			UpdateFunc: func(old, new interface{}) {
 				handleError(npc.UpdateNetworkPolicy(old.(*extnapi.NetworkPolicy), new.(*extnapi.NetworkPolicy)))

